src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginName.java

```diff
@@ -1,15 +1,48 @@
 package com.github.manolo8.darkbot.gui.plugins;
 
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.extensions.plugins.Plugin;
 import com.github.manolo8.darkbot.extensions.plugins.PluginDefinition;
+import com.github.manolo8.darkbot.extensions.plugins.PluginHandler;
+import com.github.manolo8.darkbot.extensions.plugins.PluginIssue;
+import com.github.manolo8.darkbot.gui.components.MainButton;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+
+import static com.github.manolo8.darkbot.extensions.plugins.PluginHandler.PLUGIN_OLD_PATH;
+import static com.github.manolo8.darkbot.extensions.plugins.PluginHandler.PLUGIN_UPDATE_PATH;
 
 class PluginName extends JPanel {
 
-    PluginName(PluginDefinition definition) {
-        super(new MigLayout("", "[]5px[]5px[]5px[]"));
+    private final PluginHandler pluginHandler;
+    private final Plugin plugin;
+
+    private final UpdateProgressBar progressBar;
+    private final JLabel progressLabel;
+    private final UpdateButton updateButton;
+
+    volatile boolean canReload;
+    UpdateTask task;
+
+    PluginName(Main main, Plugin plugin, UpdateProgressBar progressBar, JLabel progressLabel) {
```

Comment:
This is looking very, VERY complicated for what a "name" should be. This MUST be simplified to the minimum